### ISSUE TITLE
Accessibility Support

### ISSUE THREAD

**Issue by iiroj** [iiroj/styled-modal#2]
Our modal should always have the attribute `role="dialog"`, and optionally at least the `aria-labeledby="#modalTitle"` along with the corresponding title from a prop.

Additionally, when the modal is open, other contents should be hidden with the attributes `inert` ([link](https://github.com/WICG/inert)) and `aria-hidden="true"`. This is easier because we portal the Modal to another dom element, so this could be achieved by simply supplying a `ariaHiddenId="#App` prop to the Modal, which would then apply those to the specified element.

**Comment by iiroj** [comment]
The dialog role is included in https://github.com/iiroj/styled-modal/pull/4
